### DISCUSSION TITLE
CORS Policy Blocking JS modules

### DISCUSSION THREAD

**Discussion by Physics722** [community/community#65033] [Other Feature Feedback, Questions, & Ideas] [answered]
When I try to link a js module it comes with this error:

Access to script at '[local URL]' from origin 'null' has been blocked by CORS policy: Cross origin requests are only supported for protocol schemes: http, data, isolated-app, chrome-extension, chrome-untrusted, https, edge.

This is my HTML:
```
<!DOCTYPE html>
<html>

<head>
    <script src="JS/mainForCopy.js" type="module"></script>
</head>

<body></body>
</script>

</html>
```

This is the mainForCopy.js file: 
```
import *  as THREE from "three/src/Three.js"; 
```

But when I disable the the `import` (by commenting it) the error disappears.

How do I fix this?

**Comment by arafajs** [comment]
The error message is saying that the browser is blocking the request to the JavaScript module because it is a cross-origin request. A cross-origin request is a request from one origin (domain, protocol, and port) to another origin. In this case, the origin of the request is the local file system, and the origin of the module is the web server where the module is hosted.

To fix the CORS error, you need to make sure that the web server where the module is hosted is configured to allow cross-origin requests. This can be done by adding the following header to the response for the module:

`Access-Control-Allow-Origin: *`

**Comment by Umut-Polat-GM** [comment, marked as answer]
## The error indicates that you're trying to load a JavaScript module from a local file system (file://) and not from a supported protocol (http, https, etc.)
* Serve Files Locally: Instead of directly opening the HTML file in your browser from the local file system, set up a local development server to serve your files. This way, you can use a supported protocol (http://localhost/) and avoid CORS issues. You can use tools like Node.js with Express or Python's built-in http.server module to quickly set up a local server.
